### PR TITLE
Simplify notification clear

### PR DESCRIPTION
**Pull request by lorumic**
## Done

- Reuse code to clear a notification

### CODE REVIEW

**Review by edlerd — APPROVED** [review]
LGTM, just one comment about avoiding the shorthand if/else style.

File: src/context/notify.tsx [hunk 1]
```diff
@@ -77,13 +77,13 @@ export const NotifyProvider: FC<NotifyProviderProps> = ({ children }) => {
...
+  const clear = () => notification !== null && setNotification(null);
+
   useEffect(() => {
     if (state?.queuedNotification) {
       setDeduplicated(state.queuedNotification);
       window.history.replaceState({}, "");
-    } else {
-      notification !== null && setNotification(null);
-    }
+    } else clear();
```
[the remark this answers is not in the document]
Comment (lorumic) [reply src/context/notify.tsx:86]: I don't see how an inline else can be a source of bugs, but if it troubles you I can enclose it.

[the remark this answers is not in the document]
Comment (edlerd) [reply src/context/notify.tsx]: The most famous example is from apple: https://www.synopsys.com/blogs/software-security/understanding-apple-goto-fail-vulnerability-2/

[the remark this answers is not in the document]
Comment (lorumic) [reply src/context/notify.tsx]: Seems pretty different/not ascribable to an inline else statement. Simply put, anything can be a source of bug if you're not careful enough.

**Pull request merged**